os/board/common/crashdump.c

```diff
@@ -238,6 +317,64 @@ static int send_fsdump(void)
 	return 0;
 }
 
+static int send_extfsdump(void)
+{
+	int i = 0;
+	int fd;
+	int index;
+	uint8_t *ptr;
+	size_t size;
+	uint16_t count = 0;
+	char ext_userfs_partition[CONFIG_PATH_MAX] = { '\0' };
+	char ext_fsdump_buf[CONFIG_MTD_SMART_SECTOR_SIZE + 1];
+
+	/*Send ACK for EXTERNAL FSDUMP Handshake */
+	up_lowputc('A');
+
+	i = get_ext_userfs_part(&size, &index);
+	if (i < 0) {
+		return ERROR;
+	}
+
+	/* Send external userfs partition size to host */
+	ptr = (uint8_t *)&size;
+	for (i = 0; i < sizeof(size); i++) {
+		up_lowputc((uint8_t)*ptr);
+		ptr++;
+	}
+
+	/* If extracted size is 0, it means there is no partition available to dump */
+	if (size == 0) {
+		return 0;
+	}
+
+	snprintf(ext_userfs_partition, CONFIG_PATH_MAX, "/proc/fs/smartfs/smart1p%d/dump_psector", index);
+	fd = open(ext_userfs_partition, O_RDONLY);
+	if (fd < 0) {
+		return errno;
+	}
+
+	/* Send dummy read request, no data will actually be returned. Procfs entry automatically sends data over tty */
+	while (size) {
+		ext_fsdump_buf[1] = (uint8_t)count;
+		ext_fsdump_buf[0] = (uint8_t)(count >> 8); //Buffer now holds the sector number whose data is to be retrieved.
+		i = read(fd, ext_fsdump_buf, CONFIG_MTD_SMART_SECTOR_SIZE);
```

Comment:
Actually sector size is decided after read header of mtd sector, if it is empty, then we use this config.
We have a TODO list there, please see TODO at the start of smart_scan in smart.c
So here also TODO need to be added.

Reply:
Done